product/domain/productBasics.go

```diff
@@ -601,3 +601,15 @@ func (a Attributes) Attribute(key string) Attribute {
 	attribute := a[key]
 	return attribute
 }
+
+// AttributesByCode returns slice of attributes by given attribute codes
+func (a Attributes) AttributesByCode(codes []string) []Attribute {
```

Comment:
I guess the Attributes are key-value pairs. I would therefore recommend to call the method `AttributesByKeys`

Reply:
Otherwise please change the body of the method and actually compare the attributes code..